enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,226 @@
+
+### Drawbacks
+
+In the absence of the default ingress controller, the Route API is left unserved. The OpenShift components or workloads can create routes but no controller is available to back them up. This may be misleading for the users.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+TBD
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+N/A
+
+#### Tech Preview -> GA
+N/A
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Same as the component selection [strategy](/enhancements/installer/component-selection.md#upgrade--downgrade-strategy).
+
+### Version Skew Strategy
+N/A
+
+### Operational Aspects of API Extensions
+N/A - no api extensions are being introduced in this proposal
+
+#### Failure Modes
+N/A - no api extensions are being introduced in this proposal
+
+#### Support Procedures
+N/A - no api extensions are being introduced in this proposal
+
+## Implementation History
+N/A
+
+## Alternatives
+
+### Implement the ingress capability for the standalone OpenShift
+
+The goal of this enhancement is to satisfy the HyperShift's requirements. The implementation of the ingress capability for the standalone OpenShift brings a lot of additional complexity, unnecessary for HyperShift. The installer and the authentication operator would need to be updated to tolerate the absence of the default ingress controller.
+
+### Alternative to the cluster capability to disable the ingress operator
+
+Due to the lack of understanding of how HyperShift uses the cluster version operator to manage the ingress operator, we thought that the capability would serve only the informative purpose for the other OpenShift components like the console operator.
+That brought the idea of giving up with the capability in favor of some other cluster resource (existing or not) which could fulfill the need of informing other OpenShift components.
+Potentially that could save us from the difficult design and implementation decisions which the cluster capability implies. [The hypershift chapter](#hypershift) explains in details why the capability should be preferred over any other API.
+
+### Add Route API to the ingress capability
+
+The Route API is a fundamental part of OpenShift, a lot of OpenShift components and user workloads rely on the routes. Unlike the absence of the ingress operator which can be simulated without a dedicated capability, the removal of the route api needs the code changes in the [openshift-api-server](https://github.com/openshift/openshift-apiserver/tree/master/pkg/route).
+The impact of the Router API removal is difficult to anticipate and plan. This puts the implementation of the RFE at risk.
+In the situation when the ingress is disabled we have to choose between two evils: leave routes unserved or forbid their creation. The former appears to be the lesser evil.
```

Comment:
It seems more complicated than these two choices:

1. Ingress is enabled and the Route functions as we're used to.  Everyone's happy, except folks who wish we didn' have an ingress operator and router operand runniing.
2. Ingress is disabled:
    1. But the Route API still exists.  Components who need routing create Routes.
        1. And nothing fulfills those requests.  And the components that created the Routes feel sad.
         2. And something else fulfills those Routes, e.g. by bridging them to resources backed by the AWS load-balancer operator and its controllers.
    2. And the Route API doesn't exist either.  Components who had been using Routes have to learn to understand this situation and not create Routes.  Which may lead to customer-facing workflow changes explaining that when Ingress is disabled, the customer is on their own to plumb external clients through to the Service.

Or something like that.

Reply:
More options added. Also I tried to do the better job in convincing the reader about the option chosen.